src/main/java/in/projecteka/consentmanager/consent/ConsentManager.java

```diff
@@ -301,4 +301,53 @@ private ConsentArtefactLightRepresentation from(HIPConsentArtefactRepresentation
                 .filter(consentArtefact -> !isNotSameRequester(consentArtefact.getConsentDetail(), requesterId))
                 .switchIfEmpty(Mono.error(ClientError.consentArtefactForbidden()));
     }
+
+    public Mono<ConsentRepresentation> getConsentRepresentation(String consentId, String requesterId) {
+        return getConsentWithRequest(consentId)
+                .switchIfEmpty(Mono.error(ClientError.consentArtefactNotFound()))
+                .flatMap(consentRepresentation -> {
+                    if (isNotSameRequester(consentRepresentation.getConsentDetail(), requesterId)) {
+                        return Mono.error(ClientError.consentArtefactForbidden());
```

Comment:
You have to check whether it's in `GRANTED` state, otherwise return an error with HTTP status code 409.